apps/docs/src/app/platform/component-docs/platform-forms/platform-input/platform-input-example/platform-input-auto-complete-validation-example.component.ts

```diff
@@ -29,32 +30,35 @@ export class PlatformInputAutoCompleteValidationExampleComponent implements OnIn
     @Input()
     open: boolean;
 
-    /**
-     * The template with which to display the individual listed items.
-     * Use it by passing an ng-template with implicit content. See examples for more info.
-     */
-    @Input()
-    itemTemplate: TemplateRef<any>;
+    @ViewChild('typeahead')
+    typeahead: PopoverComponent;
 
     constructor(private fb: FormBuilder) {}
 
     ngOnInit(): void {
-        this.options = this.sportsData;
+        this.options = [];
     }
 
     filter(value: string): string[] {
         const filterValue = value.toLowerCase();
+        if (filterValue.length === 0) {
+            return [];
+        }
         return this.sportsData.filter((item: string) => item.toLowerCase().includes(filterValue));
     }
 
     onSearchChange(): void {
         this.options = this.filter(this.inputText);
-        this.open = true;
+        if (this.options.length > 0) {
+            this.typeahead.open();
+        }
+        this.typeahead.updatePopover();
```

Comment:
I would suggest to leave `[isOpen]="open"` approach and not use `this.typeahead.open()`
The issue with imperative approach that `Popover` component uses OnPush strategy as well as `popover directive` inside it. 
Particular in this case `popover.open()` just updates boolean property which is the `Input` of underlying `popover.directive`
https://github.com/SAP/fundamental-ngx/blob/34a990401692f5c4dbcfb8cbf3ccac886bb5443c/libs/core/src/lib/popover/popover.component.ts#L151-L156

`this.typeahead.open()` does not work properly since all of them use `OnPush` and this change happens outside component `change detection`.

We have to use `inputs` to make it working:
```
if (this.options.length > 0) {
   this.open = true;
} else {
    this.open = false;
}
```
In order to use `this.typeahead.open()` we have to refactor `open` method and add `changeDetectorRef.markForCheck()`

Reply:
ok I've implemented your suggestions.